yacc.go

```diff
@@ -3331,18 +3323,17 @@ type $$Parser interface {
 }
 
 type $$ParserImpl struct {
-	lookahead func() int
+	lval  $$SymType
+	stack [16]$$SymType
```

Comment:
That would be my intuition as well but Peter mentioned at lunch that the size appeared to be a significant factor (more pointers for the GC to traverse).